my-vue3-app/src/components/ReportsTable.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <p>{{ isLoading }}</p>
+  <DataTable :value="reports" :lazy="true"  v-model:filters="filters"
+    :paginator="true" :resizableColumns="true" :rows="pagination.pageSize"
+    :totalRecords="pagination.itemsTotal" :rowsPerPageOptions="[10,20,50,100]"
+    :paginatorTemplate="pag" @page="onPage($event)" filterDisplay="row"
+    @filter="onFilter($event)"
+    :globalFilterFields="['nick', 'title']"
+    >
+    <Column field="nick" header="Nick" style="min-width: 14rem" ref="nick">
+      <template #filter="{filterModel,filterCallback}">
+        <InputText type="text" v-model="filterModel.value" @keydown.enter="filterCallback()"
+        class="p-column-filter" placeholder="Search by nick"/>
+      </template>
+      <template #body="{data}">
+        {{data.nick}}
+      </template>
+    </Column>
+    <Column field="title" header="Title" style="min-width: 14rem" ref="title">
+      <template #filter="{filterModel,filterCallback}">
+        <InputText type="text" v-model="filterModel.value" @keydown.enter="filterCallback()"
+        class="p-column-filter" placeholder="Search by title"/>
+      </template>
+      <template #body="{data}">
+        {{data.title}}
+      </template>
+    </Column>
+    <Column field="id" header="ID" style="min-width: 14rem">
+      <template #body="{data}">
+        {{data.id}}
+      </template>
+    </Column>
+    <Column field="gpgSignature" header="GPG Signature" style="min-width: 14rem">
+      <template #body="{data}">
+        {{data.gpgSignature}}
+      </template>
+    </Column>
+  </DataTable>
+</template>
+
+<script lang="ts">
+import DataTable from 'primevue/datatable';
+import {
+  IS_LOADING, PAGINATION, REPORTS_MODULE, State as ReportsState,
+} from '@/store/reports';
+import { FETCH_REPORTS } from '@/store/reports/actions';
+import { defineComponent } from 'vue';
+import { mapActions, mapState } from 'vuex';
+import Column from 'primevue/column';
+import { Report } from '@/services/api';
+import { debounce } from 'lodash-es';
+
+type FetchParams = {
+  page: number;
+  pageSize: number;
+  filters: {
+    nick: {
+      value: string|undefined;
+      type: string;
```

Comment:
So we decided to change our API for the component

Reply:
+